File: src/main/java/name/seguri/java/leetcode/_226/Solution.java
package name.seguri.java.leetcode._226;

class Solution {
  TreeNode invertTree(TreeNode node) {
    if (node == null) {
      return null;
    }
    TreeNode temp = node.left;
    node.left = node.right;
    node.right = temp;
    invertTree(node.left);
    invertTree(node.right);
    return node;
  }

  TreeNode invertTree(int... values) {
    var root = TreeNode.of(values);
    return invertTree(root);
  }
}

File: src/main/kotlin/name/seguri/kotlin/leetcode/_226/Solution.kt
package name.seguri.kotlin.leetcode._226

class Solution {
  fun invertTree(node: TreeNode?): TreeNode? {
    if (node == null) {
      return null
    }
    val temp = node.left
    node.left = node.right
    node.right = temp
    invertTree(node.left)
    invertTree(node.right)
    return node
  }

  fun invertTree(vararg values: Int): TreeNode? {
    var root = TreeNode.of(*values)
    return invertTree(root)
  }
}
